tests/components/airly/test_config_flow.py

```diff
@@ -35,69 +34,62 @@ async def test_show_form(hass):
-            HTTP_FORBIDDEN, {"message": "Invalid authentication credentials"}
+    aioclient_mock.get(
+        API_KEY_VALIDATION_URL,
+        exc=AirlyError(
+            HTTP_UNAUTHORIZED, {"message": "Invalid authentication credentials"}
         ),
-    ):
+    )
 
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
-        )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+    )
 
-        assert result["errors"] == {"base": "invalid_api_key"}
+    assert result["errors"] == {"base": "invalid_api_key"}
 
 
-async def test_invalid_location(hass):
+async def test_invalid_location(hass, aioclient_mock):
     """Test that errors are shown when location is invalid."""
-    with patch(
-        "airly._private._RequestsHandler.get",
-        return_value=json.loads(load_fixture("airly_no_station.json")),
-    ):
+    aioclient_mock.get(
+        API_KEY_VALIDATION_URL, text=load_fixture("airly_valid_station.json")
+    )
+    aioclient_mock.get(API_POINT_URL, text=load_fixture("airly_no_station.json"))
 
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
-        )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+    )
 
-        assert result["errors"] == {"base": "wrong_location"}
+    assert result["errors"] == {"base": "wrong_location"}
 
 
-async def test_duplicate_error(hass):
+async def test_duplicate_error(hass, aioclient_mock):
     """Test that errors are shown when duplicates are added."""
+    aioclient_mock.get(API_POINT_URL, text=load_fixture("airly_valid_station.json"))
+    MockConfigEntry(domain=DOMAIN, unique_id="123-456", data=CONFIG).add_to_hass(hass)
 
-    with patch(
-        "airly._private._RequestsHandler.get",
-        return_value=json.loads(load_fixture("airly_valid_station.json")),
-    ):
-        MockConfigEntry(domain=DOMAIN, unique_id="123-456", data=CONFIG).add_to_hass(
-            hass
-        )
-
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
-        )
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+    )
 
-        assert result["type"] == "abort"
-        assert result["reason"] == "already_configured"
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
 
 
-async def test_create_entry(hass):
+async def test_create_entry(hass, aioclient_mock):
     """Test that the user step works."""
+    aioclient_mock.get(
+        API_KEY_VALIDATION_URL, text=load_fixture("airly_valid_station.json")
+    )
+    aioclient_mock.get(API_POINT_URL, text=load_fixture("airly_valid_station.json"))
 
-    with patch(
```

Comment:
Please patch the integration setup functions here. That's standard procedure for all tests with entry create result.